File: backend/backend-java/src/main/java/com/cayena/backend/entities/Supplier.java
package com.cayena.backend.entities;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import java.time.LocalDateTime;
@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Entity
public class Supplier {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @JsonProperty("suplier_id")
    private Long suplierId;
    private String name;
    @JsonProperty("date_of_creation")
    private LocalDateTime dateOfCreation;
    @JsonProperty("date_of_the_last_update")
    private LocalDateTime dateOfTheLastUpdate;
}

File: backend/backend-kotlin/src/main/kotlin/com/cayena/backendkotlin/dominio/Supplier.kt
package com.cayena.backendkotlin.dominio

import com.fasterxml.jackson.annotation.JsonProperty
import java.time.LocalDateTime
import javax.persistence.*

@Entity
data class Supplier (
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @JsonProperty("supplier_id")
    val supplierId: Long,
    @JsonProperty("name_supplier")
    val nameSupplier: String,
    @JsonProperty("date_of_creation")
    val dateOfCreation: LocalDateTime? = LocalDateTime.now(),
    @JsonProperty("date_of_the_last_update")
    val dateOfTheLastUpdate: LocalDateTime? = LocalDateTime.now()
)

